From 13710e6cf98843ceb811c0f2c1e384b1c1beb3db Mon Sep 17 00:00:00 2001
From: Ravi Patel <ravi.patel@xilinx.com>
Date: Tue, 3 Dec 2019 01:03:52 -0800
Subject: [PATCH 0066/1851] firmware: xilinx: xilinx: Add IOCTL ids for probe
 counter

commit df5564e4f9e2825198d07268d59b4701240a30cb from
https://github.com/Xilinx/linux-xlnx.git

Add IOCTL Ids for probe counters to access the flexnoc APM registers.

Signed-off-by: Ravi Patel <ravi.patel@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Rajan Vaja <rajan.vaja@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/firmware/xilinx/zynqmp-debug.c | 3 ++-
 drivers/firmware/xilinx/zynqmp.c       | 2 ++
 include/linux/firmware/xlnx-zynqmp.h   | 3 +++
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/firmware/xilinx/zynqmp-debug.c b/drivers/firmware/xilinx/zynqmp-debug.c
index 1ba7bfa50590..1f8dc360a249 100644
--- a/drivers/firmware/xilinx/zynqmp-debug.c
+++ b/drivers/firmware/xilinx/zynqmp-debug.c
@@ -314,7 +314,8 @@ static int process_api_request(u32 pm_id, u64 *pm_api_arg, u32 *pm_api_ret)
 			     pm_api_arg[1] == IOCTL_GET_PLL_FRAC_MODE ||
 			     pm_api_arg[1] == IOCTL_GET_PLL_FRAC_DATA ||
 			     pm_api_arg[1] == IOCTL_READ_GGS ||
-			     pm_api_arg[1] == IOCTL_READ_PGGS))
+			     pm_api_arg[1] == IOCTL_READ_PGGS ||
+			     pm_api_arg[1] == IOCTL_PROBE_COUNTER_READ))
 			sprintf(debugfs_buf, "IOCTL return value: %u\n",
 				pm_api_ret[1]);
 		break;
diff --git a/drivers/firmware/xilinx/zynqmp.c b/drivers/firmware/xilinx/zynqmp.c
index c4e0abe31645..e18595ecddac 100644
--- a/drivers/firmware/xilinx/zynqmp.c
+++ b/drivers/firmware/xilinx/zynqmp.c
@@ -531,6 +531,8 @@ static inline int zynqmp_is_valid_ioctl(u32 ioctl_id)
 	case IOCTL_ULPI_RESET:
 	case IOCTL_SET_BOOT_HEALTH_STATUS:
 	case IOCTL_AFI:
+	case IOCTL_PROBE_COUNTER_READ:
+	case IOCTL_PROBE_COUNTER_WRITE:
 		return 1;
 	default:
 		return 0;
diff --git a/include/linux/firmware/xlnx-zynqmp.h b/include/linux/firmware/xlnx-zynqmp.h
index 49c66d2ccbd7..880493692e62 100644
--- a/include/linux/firmware/xlnx-zynqmp.h
+++ b/include/linux/firmware/xlnx-zynqmp.h
@@ -167,6 +167,9 @@ enum pm_ioctl_id {
 	/* Set healthy bit value*/
 	IOCTL_SET_BOOT_HEALTH_STATUS,
 	IOCTL_AFI,
+	/* Probe counter read/write */
+	IOCTL_PROBE_COUNTER_READ,
+	IOCTL_PROBE_COUNTER_WRITE,
 };
 
 enum pm_query_id {
-- 
2.31.1

